pgx/src/datum/array.rs

```diff
@@ -456,4 +436,9 @@ where
     fn type_oid() -> u32 {
         unsafe { pg_sys::get_array_type(T::type_oid()) }
     }
+
+    #[inline]
+    fn is_compatible_with(other: pg_sys::Oid) -> bool {
```

Comment:
I notice this pattern in several implementations, I wonder if we could provide a default implementation in the trait?